plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -74,17 +75,67 @@ func loadStrategies(strategiesFile string) (*strategies, error) {
 }
 
 func (h *strategyStore) parseStrategies(strategies *strategies) {
-	h.defaultStrategy = &defaultStrategy
+	h.defaultStrategy = &defaultStrategyResponse
 	if strategies == nil {
 		h.logger.Info("No sampling strategies provided, using defaults")
 		return
 	}
 	if strategies.DefaultStrategy != nil {
-		h.defaultStrategy = h.parseStrategy(strategies.DefaultStrategy)
+		h.defaultStrategy = h.parseDefaultStrategy(strategies.DefaultStrategy)
 	}
+
 	for _, s := range strategies.ServiceStrategies {
 		h.serviceStrategies[s.Service] = h.parseServiceStrategies(s)
+
+		// Merge with the default operation strategies, because only merging with
+		// the default strategy has no effect on service strategies (the default strategy
+		// is not merged with and only used as a fallback).
+		opS := h.serviceStrategies[s.Service].OperationSampling
+		if opS == nil {
+			// It has no use to merge, just reference the default settings.
+			h.serviceStrategies[s.Service].OperationSampling = h.defaultStrategy.OperationSampling
+			continue
+		}
+		if h.defaultStrategy.OperationSampling == nil ||
+			h.defaultStrategy.OperationSampling.PerOperationStrategies == nil {
+			continue
+		}
+		opS.PerOperationStrategies = mergePerOperationSamplingStrategies(
+			opS.PerOperationStrategies,
+			h.defaultStrategy.OperationSampling.PerOperationStrategies)
+	}
+}
+
+// mergePerOperationStrategies merges two operation strategies a and b, where a takes precedence over b.
+func mergePerOperationSamplingStrategies(
+	a, b []*sampling.OperationSamplingStrategy,
+) []*sampling.OperationSamplingStrategy {
+	// Guess the size of the slice of the two merged.
+	merged := make([]*sampling.OperationSamplingStrategy, 0, (len(a)+len(b))/4*3)
```

Comment:
The idea is that, if a service already defines a strategy for a method, the default strategy should not override this and simply use the strategy defined for the service instead of the default.